listenbrainz/webserver/static/js/src/PinnedRecordingCard.test.tsx

```diff
@@ -0,0 +1,327 @@
+import * as React from "react";
+import { mount, shallow } from "enzyme";
+
+import PinnedRecordingCard, {
+  PinnedRecordingCardProps,
+} from "./PinnedRecordingCard";
+import * as utils from "./utils";
+import APIServiceClass from "./APIService";
+import GlobalAppContext from "./GlobalAppContext";
+
+// Font Awesome generates a random hash ID for each icon everytime.
+// Mocking Math.random() fixes this
+// https://github.com/FortAwesome/react-fontawesome/issues/194#issuecomment-627235075
+jest.spyOn(global.Math, "random").mockImplementation(() => 0);
+
+const user = {
+  id: 1,
+  name: "name",
+  auth_token: "auth_token",
+};
+
+const globalProps = {
+  APIService: new APIServiceClass(""),
+  currentUser: user,
+  spotifyAuth: {},
+  youtubeAuth: {},
+};
+
+const pinnedRecording: PinnedRecording = {
+  blurb_content: "I LOVE",
+  created: 1111111111,
+  pinned_until: 9999999999,
+  row_id: 1,
+  recording_mbid: "98255a8c-017a-4bc7-8dd6-1fa36124572b",
+  track_metadata: {
+    artist_name: "Rick Astley",
+    track_name: "Never Gonna Give You Up",
+  },
+};
+
+const expiredPinnedRecording: PinnedRecording = {
+  ...pinnedRecording,
+  pinned_until: 1111122222,
+};
+
+const props: PinnedRecordingCardProps = {
+  userName: user.name,
+  PinnedRecording: pinnedRecording,
+  isCurrentUser: true,
+  newAlert: () => {},
+};
+
+describe("PinnedRecordingCard", () => {
+  it("renders correctly", () => {
+    const wrapper = mount<PinnedRecordingCard>(
+      <PinnedRecordingCard {...props} />
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
+});
+
+describe("determineIfCurrentlyPinned", () => {
+  it("returns true when pinned_until > now", () => {
+    const wrapper = mount<PinnedRecordingCard>(
+      <PinnedRecordingCard {...props} />
+    );
+    const instance = wrapper.instance();
+    expect(instance.determineIfCurrentlyPinned()).toBeTruthy();
```

Comment:
nit
```suggestion
    expect(instance.determineIfCurrentlyPinned()).toBe(true);
```

we know that this function always returns a bool , so tightening up the assertion would be a good idea.